src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1625,7 +1640,16 @@ private static bool QueueUserWorkItemHelper(WaitCallback callBack, Object state,
                 try { }
                 finally
                 {
-                    QueueUserWorkItemCallback tpcallBack = new QueueUserWorkItemCallback(callBack, state, compressStack, ref stackMark);
+                    ExecutionContext context = compressStack && !ExecutionContext.IsFlowSuppressed() ?
+                                                ExecutionContext.Capture(ref stackMark, ExecutionContext.CaptureOptions.IgnoreSyncCtx | ExecutionContext.CaptureOptions.OptimizeDefaultCase) :
+                                                ExecutionContext.Default;
+
+                    QueueUserWorkItemCallback tpcallBack = (context == ExecutionContext.Default) ?
+                                                (state == null ? 
+                                                    new QueueUserWorkItemCallback(callBack) : // default or null context, no state 
+                                                    new QueueUserWorkItemCallbackWithState(callBack, state)) : // default or null context
```

Comment:
You can't suppress flow the in CoreCLR, and `UnsafeQueueUserWorkItem` isn't exposed so is this change actually worth it as you can't get to the non-execution context version from the threadpool?